### DISCUSSION TITLE
Как нормально вызвать set с большим количеством field ?

### DISCUSSION THREAD

**Discussion by simplepersonru** [totumonline/totum-ru-issues-and-discussions#415] [Как это сделать?] [answered]
несколько раз перечитал https://ru.docs.totum.online/functions#fn-set касательно аргумента field. Попробовал вставлять в field переменные, созданные через jsonCreate, rowCreate. Пробовал делать список ассоциативных пар. Каждый раз ошибка что Totum не понимает такой синтаксис. Единственный вариант - раздувать текст вширь, добавляя все новые и новые field в вызов функции set. (в какой-то момент там сломалась правая граница и я перестал видеть последний кусок функции и до него стало не промотать, но это уже другая история).

Возможные решения (если конечно под это где-то уже что-то не сделано):

1. позволить переносить на новые строки части вызова функции. Например чтобы новый аргумент можно было присваивать с новой строки. Сейчас это ошибка так делать
2. Добавить в функцию set вариант с аргументом, который принимает список ассоциативных пар (тогда можно будет хотя бы обьединять списки на разных строчках и в итоге присвоить финальный

**Comment by poluninatm** [comment]
Опишите задачу.
Посмотрите в сторону параметра fields функции setListExtended. Обратите
внимание на то, что это list-функция в отличие от set.

Вс, 16 июля 2023 г. в 00:04, ArtemBabaev ***@***.***>:

**Reply by simplepersonru** [reply]
Пока описывал понял, что можно решить по другому - через обязательные поля и значения по умолчанию.

Задача была в создании расчетной в цикле для таблицы циклов. Я делал через отдельную временную таблицу с почти такими же полями (не все), но вот выше написал - по другому сделаю

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Перенос строк кода известная задача. Глюк неправильного определения ширины кодового поля в меньшую сторону - известный глюк. Неудобство использования функции с большим количеством множественных параметров - известное неудобство. Волевым усилием я решаю .... ничего с ними не делать.
